RFC-0001.md

```diff
@@ -0,0 +1,230 @@
+true table.
+
+One benefit of creating the sample tables as an iceberg table is that we can re-use the
+infrastructure of the iceberg connector to read the tables. All that is necessary is to properly
+pass a correctly constructued `IcebergTableHandle` and then pass it to the necessary functions. 
+
+#### Time Travel and Schema Updates
+
+The iceberg connector supports time travel and schema updates to the columns and partitions of the
+table. The sample table should undergo the same schema updates whenever the real table does. The
+implementation should perform the same schema updates that the real table does.
+
+In the implementation, this means that when the connector calls `Metadata#renameColumn`,
+`dropColumn`, etc then the operation should run on both the real and sample table.
+
+#### Correctly Generating Stats through ANALYZE
+
+When using samples, generally all of the stats except for distinct values can map 1:1 onto the stats
+for the full table. So, there needs to be some additional logic to estimate the number of distinct
+values in the real dataset based on the sample[^2].
+
+A recent PR, #20993, introduced the ability for connectors to specify the function used for a
+specific column statistic aggregation. When analyzing a sample table, we can replace the
+`approx_distinct` function with a separate function which uses one of the sample NDV estimators.
+We'll then write the distinct values from the result of that function into the iceberg table.
+Initially we can use the Chao Estimator[^1] as it's easy to implement, but can replace it with
+different estimators as we see fit.
+
+#### Statistics for Partitioned Tables
+
+Currently, iceberg does not fully support support partition-level statistics[^3]. Once partitions
+statistics are officially released, the iceberg connector should be updated to support collecting
+and reading the partition-level stats. As long as the sample table creation code ensures the sample
+table's layout is the exact same as the true table we should not need to change the stats
+calculation code path in the connector's implementation of `getTableStatistics`.
+
+#### Sample Metadata
+
+For external systems to determine when and how to update the sample, additional metadata may
+need to be stored, such as the last snapshot ID of the table used to create or update the sample, 
+or the sampling method and/or source used.
+
+As long as the metadata stored is limited to something small, say, less than 1KiB, then it is
+probably fine to store this information inside of the iceberg's table properties which is a simple
+`map<string, string>`.
+
+#### Sample Maintenance
+
+Sample maintenance can be performed through a set of SQL statements which replace records in the
+sample with updates, deletes, or inserts that occur on the real table. With the introduction of
+Iceberg's changelog table in #20937 we can efficiently update the sample after the table has been
+updated without needing to re-sample the entire table.
```

Comment:
Iceberg changelog  table currently do not support V2 row level deletion. It seems to be a big problem for incremental maintenance.